src/Microsoft.AspNet.OData/Routing/ODataActionSelector.cs

```diff
@@ -106,5 +112,80 @@ private static HttpResponseMessage CreateErrorResponse(HttpRequestMessage reques
 
             return request.CreateErrorResponse(statusCode, error);
         }
+
+        private static bool ActionParametersMatchRequest(HttpActionDescriptor action, HttpControllerContext context)
+        {
+            var parameters = action.GetParameters();
+            var routeData = context.RouteData;
+            var conventionsStore = context.Request.ODataProperties().RoutingConventionsStore;
+            var matchedBody = false;
+            var route = routeData.Route as ODataRoute;
+            var routePrefix = route?.RoutePrefix;
+            var availableKeys = routeData.Values.Keys
+                .Where(k => routePrefix != "{" + k + "}"
+                    && k != ODataRouteConstants.Action
+                    && k != ODataRouteConstants.Controller
+                    && k != ODataRouteConstants.ODataPath)
+                .Select(k => k.ToUpperInvariant())
```

Comment:
>ToUpperInvariant [](start = 31, length = 16)

In AspNetCore we use ToLowerInvariant.  Might be better to be consistent in case we write common routines/merge code later?